src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -822,6 +823,51 @@ public void testPrintSuppressionOnMoreThanOneFile() throws Exception {
         Main.main("-s", "2:4", getPath(""), getPath(""));
     }
 
+    @Test
+    public void testGenerateXpathSuppressionOption() throws Exception {
+        final String expected = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+            + "<!DOCTYPE suppressions PUBLIC\n"
+            + "    \"-//Puppy Crawl//DTD Suppressions Xpath Experimental 1.2//EN\"\n"
+            + "    \"http://checkstyle.sourceforge.net/dtds/"
+            + "suppressions_1_2_xpath_experimental.dtd\">\n"
+            + "<suppressions>\n"
+            + "<suppress-xpath\n"
+            + "       files=\"InputMainComplexityOverflow.java\"\n"
+            + "       checks=\"com.puppycrawl.tools.checkstyle.checks.blocks.LeftCurlyCheck\"\n"
```

Comment:
hmm, Checks that are in our library should be referenced as simply "LeftCurly".
as you can see at examples at https://checkstyle.org/config_filters.html#SuppressionXpathFilter

I am ok to move this to separate issue, to let us merge this PR quicker.